src/EFCore.PG/Query/ExpressionTranslators/Internal/NpgsqlStringMethodTranslator.cs

```diff
@@ -425,6 +431,30 @@ public NpgsqlStringMethodTranslator(NpgsqlTypeMappingSource typeMappingSource, I
                 _typeMappingSource.FindMapping(typeof(string[])));
         }
 
+        if (method == ToDate)
+        {
+            return _sqlExpressionFactory.Function(
+                "to_date",
+                new[] { arguments[1], arguments[2] },
+                nullable: true,
+                argumentsPropagateNullability: new[] { true, false },
+                typeof(DateOnly?),
+                _typeMappingSource.FindMapping(typeof(DateOnly?))
```

Comment:
Type mappings are oblivious of nullability, so it's better to remove the question mark here.